src/mscorlib/shared/System/IO/PathHelper.Windows.cs

```diff
@@ -233,26 +221,26 @@ private static string TryExpandShortFileName(ref StringBuffer outputBuffer, stri
                 }
 
                 // We now need to strip out any added characters at the front of the string
-                if (bufferToUse.SubstringEquals(originalPath, rootDifference, newLength))
+                if (bufferToUse.AsSpan().Slice(rootDifference).Equals(originalPath.AsReadOnlySpan()))
                 {
                     // Use the original path to avoid allocating
                     returnValue = originalPath;
                 }
                 else
                 {
-                    returnValue = bufferToUse.Substring(rootDifference, newLength);
+                    returnValue = new string(bufferToUse.AsSpan().Slice(rootDifference, newLength));
                 }
 
                 return returnValue;
             }
             finally
             {
-                inputBuffer.Free();
+                inputBuilder.Dispose();
             }
         }
 
         // Helper method to workaround lack of operator ? support for ref values
-        private static ref StringBuffer Choose(bool condition, ref StringBuffer s1, ref StringBuffer s2)
```

Comment:
You should not need this method anymore. The current C# compiler used for coreclr/corefx should have support for ?: for ref values. I think it should be something like:
```
ref ValueStringBuilder bufferToUse = ref (success ? ref outputBuilder : ref inputBuilder);
```